# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelper.h
# Commit: 3d201ea88341
# Full Hash: 3d201ea8834161911890a06bae578b64d18eefa9
# Author: Kelsey Gilbert <jgilbert@mozilla.com>
# Date: 2021-12-15 09:11:58
# Regressor Bug: 1746111
# File Overlap Count: 1
# Description:
#   Bug 1746111 - (Re)serialize WebRTC frame SurfaceDescriptorBuffers for convert-and-upload on WebGLParent. r=gfx-reviewers,nika,lsalzman
#   
#   Also generally de-inline (WebGLParent::TexImage not DispatchCommand)
#   TexImage(SurfaceDescriptor).
#   
# ==============================================================================

diff -r a87211c8eedd -r 3d201ea88341 gfx/gl/GLBlitHelper.h
--- a/gfx/gl/GLBlitHelper.h	Wed Dec 15 01:49:11 2021 +0000
+++ b/gfx/gl/GLBlitHelper.h	Wed Dec 15 01:54:21 2021 +0000
@@ -41,8 +41,10 @@
 namespace layers {
 class Image;
 class GPUVideoImage;
+struct PlanarYCbCrData;
 class PlanarYCbCrImage;
 class SurfaceDescriptor;
+class SurfaceDescriptorBuffer;
 
 #ifdef XP_WIN
 class D3D11ShareHandleImage;
@@ -172,8 +174,8 @@
   const DrawBlitProg* CreateDrawBlitProg(const DrawBlitProg::Key& key) const;
 
  public:
-  bool BlitImage(layers::PlanarYCbCrImage* yuvImage,
-                 const gfx::IntSize& destSize, OriginPos destOrigin);
+  bool BlitPlanarYCbCr(const layers::PlanarYCbCrData&,
+                       const gfx::IntSize& destSize, OriginPos destOrigin);
 #ifdef MOZ_WIDGET_ANDROID
   // Blit onto the current FB.
   bool BlitImage(layers::SurfaceTextureImage* stImage,